WOS-167 Title and Content components: added new component properties …

AdrianStaskiewicz opened the pull request:
…- color and shadow

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## Screenshots (if appropriate)

## Upgrade notes (if appropriate)
<!-- What changes user have to do in order to migrate from the previous version to the version with this feature -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue) labeled with `bug`
- [ ] New feature (non-breaking change which adds functionality) labeled with `enhancement`
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [code standards](../CONTRIBUTING.md) of this project.
- [ ] My change requires updating the documentation. I have updated the documentation accordingly.

AdrianStaskiewicz committed:
#165 Added icon position property to the button component adn adjuste… (#166)

* #165 Button component: added possibility to define two icons (left and right)

Rafal-DS committed:
#145 created extensions module and moved tableservice and tableview i… (#167)

* #145 created extensions module and moved tableservice and tableview into it

* #145 updated readme file

AdrianStaskiewicz committed:
#163 Added new button modifier option - inverted button (#169)

#163 Button component:  added new modifier option - inverted button

AdrianStaskiewicz committed:
#81 Added new breadcrumb element property - icon (#170)

#81 Breadcrumb property: added new breadcrumb element property - icon

AdrianStaskiewicz committed:
WOS-167 Title and Content components: added new component properties - color and shadow

AdrianStaskiewicz committed:
WOS-167 Added test

AdrianStaskiewicz committed:
WOS-167 Fixed property name

AdrianStaskiewicz committed:
WOS-167 Reimplemented solution WiP

Rafal-DS committed:
#164 added alignment config option for level item component (#171)

* #164 added alignment config option for level item component

* #164 now the alignment works in nested containers too

* #164 unused constant

* #164 optimize imports

* #164 import order

AdrianStaskiewicz committed:
WOS-167 Small code improvement

andrechern committed:
WOS-168 Content Text weight (#173)

AdrianStaskiewicz committed:
WOS-167 Small code improvement

AdrianStaskiewicz committed:
WOS-167 Added default text color value for title

AdrianStaskiewicz committed:
Merge remote-tracking branch 'origin/develop' into WOS-167-color-variants-for-title-and-content

# Conflicts:
#	application/backend/src/main/java/pl/ds/bulma/components/models/ContentComponent.java
#	application/backend/src/main/resources/apps/bulma/components/content/content.html
#	application/backend/src/main/resources/apps/bulma/components/content/dialog/.content.json

Rafal-DS commented on application/backend/src/main/java/pl/ds/bulma/components/helpers/ColorService.java:
```diff
@@ -0,0 +1,100 @@
+  }
+
+  private List<String> getColorsListFromResource(Resource resource, String resourcePath) {
+    if (resource != null) {
```
How about an early exit to reduce nesting?
if (resource == null) {
return new ArrayList<>();
}

Rafal-DS commented on application/backend/src/main/java/pl/ds/bulma/components/models/ContentComponent.java:
```diff
@@ -16,13 +16,16 @@
+import pl.ds.bulma.components.helpers.ColorService;
 
 @Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
 public class ContentComponent {
```
Adding bold to text removes the color:
<img width="271" alt="Screenshot 2023-04-28 at 16 00 29" src="https://user-images.githubusercontent.com/121929316/235168527-4636f5b7-a189-4053-8eff-3010d21e812d.png">

paweljozwicki replied:
Removing colors by adding bold, underlining, etc. is a known issue. Let's go ahead and resolve it in a separate issue.

Rafal-DS commented on application/backend/src/main/java/pl/ds/bulma/components/models/TitleComponent.java:
```diff
@@ -59,6 +61,28 @@ public class TitleComponent {
+
+  @Inject
+  @Default(values = StringUtils.EMPTY)
+  private String shadeBw;
```
What does "Bw" mean?

AdrianStaskiewicz replied:
black and white ;)

Rafal-DS commented on application/backend/src/main/resources/apps/bulma/components/content/dialog/.content.json:
```diff
@@ -27,6 +27,80 @@
+        "sling:resourceType": "wcm/dialogs/components/include",
+        "path": "/apps/bulma/components/common/text/color"
+      },
+      "shadeContainer": {
```
I don't see this any shade container in config dialog:
<img width="726" alt="Screenshot 2023-04-28 at 16 23 52" src="https://user-images.githubusercontent.com/121929316/235174403-d94b640f-95f3-4714-b99d-cc9239c74957.png">

AdrianStaskiewicz committed:
WOS-167 Defined default values for title and content component

paweljozwicki committed:
Merge remote-tracking branch 'origin/develop' into WOS-167-color-variants-for-title-and-content

paweljozwicki committed:
Resolve conflicts

Rafal-DS committed:
Wos 185 title with subtitle (#188)

* WOS-185 added "add subtitle" checkbox

* WOS-185 subtitle config display

* WOS-185 fix tests

* WOS-185 added sly test

* WOS-185 fixed description

AdrianStaskiewicz committed:
WSIO-45 Navbar: added component template (#187)

* WSIO-45 Navbar: added component template

* WSIO-45 Navbar: applied changes requested during PR review

martyna-ds committed:
168 rte text color extension (#175)

* #165 Added icon position property to the button component adn adjuste… (#166)

* #165 Button component: added possibility to define two icons (left and right)

* #168 Add initial files

* #168 Add configuration files

* #168 Add color extension - draft, correct pom.xml

* #145 created extensions module and moved tableservice and tableview i… (#167)

* #145 created extensions module and moved tableservice and tableview into it

* #145 updated readme file

* #163 Added new button modifier option - inverted button (#169)

#163 Button component:  added new modifier option - inverted button

* #81 Added new breadcrumb element property - icon (#170)

#81 Breadcrumb property: added new breadcrumb element property - icon

* #168 Correct configuration

* #168 Add new bulma-color module

* #168 Add new bulma-color module

* #168 Toggiling state

* #168 ColorDialog

* #164 added alignment config option for level item component (#171)

* #164 added alignment config option for level item component

* #164 now the alignment works in nested containers too

* #164 unused constant

* #164 optimize imports

* #164 import order

* WOS-168 Content Text weight (#173)

* #168 Colors list - dynamic css creation

* #168 Colors list - dynamic css creation

* Wos 173 apps to libs (#180)

* WOS-173 move apps -> libs

* WOS-173 paths in sources apps -> libs

* WOS-178 init content for hero component (#181)

* WOS-173 move apps -> libs

* WOS-173 paths in sources apps -> libs

* WOS-178 init content for hero component

* WOS-176 added container style config to navbar component (#182)

* WOS-173 move apps -> libs

* WOS-173 paths in sources apps -> libs

* WOS-176 added container style config to navbar component

* WOS-176 change after review

* #170 added vertical alignment for level item component (#177)

* #168 Move color extension to extensions folder

* WOS-170 merge fix

* WOS-170 one more fix

* #168 Change /apps to /libs folder name

* #168 Move configuration to libs in backend

* #168 Move rte configuration to common

* #168 Corrections after pr, bulma fix

---------

Co-authored-by: AdrianStaskiewicz <34780625+AdrianStaskiewicz@users.noreply.github.com>
Co-authored-by: Rafal-DS <121929316+Rafal-DS@users.noreply.github.com>
Co-authored-by: andrechern <119608077+andrechern@users.noreply.github.com>
Co-authored-by: Rafał Ziemniewski <rafal.ziemniewski@ds.pl>
Co-authored-by: Paweł Jóźwicki <pawel.jozwicki@ds.pl>

paweljozwicki committed:
Update minor version (#189)

paweljozwicki committed:
Merge remote-tracking branch 'origin/main' into develop

# Conflicts:
#	application/backend/pom.xml
#	application/extensions/bulma-table-service/pom.xml
#	application/extensions/bulma-table-view/pom.xml
#	application/extensions/pom.xml
#	application/frontend/pom.xml
#	application/pom.xml
#	content/pom.xml
#	distribution/pom.xml
#	pom.xml
#	tests/content/pom.xml
#	tests/end-to-end/pom.xml
#	tests/pom.xml

AdrianStaskiewicz committed:
Merge remote-tracking branch 'origin/develop' into WOS-167-color-variants-for-title-and-content

# Conflicts:
#	application/backend/src/main/java/pl/ds/bulma/components/models/TitleComponent.java

AdrianStaskiewicz committed:
WOS-167 Title: fixed some code formatting

AdrianStaskiewicz committed:
Merge remote-tracking branch 'origin/main' into WOS-167-color-variants-for-title-and-content

# Conflicts:
#	application/backend/pom.xml
#	application/backend/src/main/java/pl/ds/bulma/components/models/TitleComponent.java
#	application/backend/src/main/resources/libs/bulma/components/title/dialog/.content.json
#	application/extensions/bulma-color/pom.xml
#	application/extensions/bulma-table-service/pom.xml
#	application/extensions/bulma-table-view/pom.xml
#	application/extensions/pom.xml
#	application/frontend/pom.xml
#	application/pom.xml
#	content/pom.xml
#	distribution/pom.xml
#	pom.xml
#	tests/content/pom.xml
#	tests/end-to-end/pom.xml
#	tests/pom.xml

Rafal-DS commented on application/backend/src/main/resources/libs/bulma/components/content/dialog/.content.json:
```diff
@@ -28,6 +28,91 @@
+      "color": {
+        "sling:resourceType": "wcm/dialogs/components/include",
+        "include": {
+          "sling:resourceSuperType": "/libs/bulma/components/common/text/color",
```
Someone told me that, we should write paths like this:
"bulma/components/common/text/color"
not like this
"/libs/bulma/components/common/text/color"
so that the resource can be overwritten.

AdrianStaskiewicz committed:
Merge remote-tracking branch 'origin/main' into WOS-167-color-variants-for-title-and-content

# Conflicts:
#	application/backend/src/main/java/pl/ds/bulma/components/models/ContentComponent.java

AdrianStaskiewicz committed:
WOS-167 Title and Content: added some code improvements

paweljozwicki commented on application/backend/src/main/java/pl/ds/bulma/components/models/TitleComponent.java:
```diff
@@ -86,6 +110,12 @@ private void init() {
     subtitleClasses = subtitleClassList.toArray(new String[]{});
+
+    ColorService colorService
+            = new ColorService(resource, "/libs/bulma/components/common/text/color",
```
You can change the path to a relative one. It will give the possibility to overwrite it in the project easily.

paweljozwicki commented on application/backend/src/main/java/pl/ds/bulma/components/helpers/ColorService.java:
```diff
@@ -0,0 +1,100 @@
+                      String shadeBw, String shadeGrey, String shadeRest) {
+    List<String> colorsList = getColorsListFromResource(resource, resourcePath);
+
+    this.colorBw = getElementsFromListByPrefix(colorsList, "bw_");
```
I'm wondering if we could read the prefix (shade_) and map colors by it. Then we could find shades by this prefix. It should be more elastic. I mean, in the project we could overwrite the whole colors and shades with something totally different.

paweljozwicki replied:
Accepted for now. We can refactor it later.

AdrianStaskiewicz committed:
WOS-190 Title: applied change requested during PR review
